core/store/src/trie/shard_tries.rs

```diff
@@ -302,6 +387,148 @@ impl ShardTries {
+                let _timer = metrics::MAKE_STATE_SNAPSHOT_ELAPSED.start_timer();
+                let mut state_snapshot_lock = self.0.state_snapshot.write().unwrap();
+                if let Some(state_snapshot) = &*state_snapshot_lock {
+                    if &state_snapshot.prev_block_hash == prev_block_hash {
+                        tracing::warn!(target: "state_snapshot", ?prev_block_hash, "Requested a state snapshot but that is already available");
+                        return Ok(());
+                    } else {
+                        let prev_block_hash = state_snapshot.prev_block_hash;
+                        // Drop Store before deleting the underlying data.
+                        *state_snapshot_lock = None;
+                        self.delete_state_snapshot(
+                            &prev_block_hash,
+                            home_dir,
+                            hot_store_path,
+                            state_snapshot_subdir,
+                        );
+                    }
+                }
+
+                let storage = checkpoint_hot_storage_and_cleanup_columns(
+                    &self.0.store,
+                    &Self::get_state_snapshot_base_dir(
+                        prev_block_hash,
+                        home_dir,
+                        hot_store_path,
+                        state_snapshot_subdir,
+                    ),
+                    Some(vec![
+                        // Keep DbVersion and BlockMisc, otherwise you'll not be able to open the state snapshot as a Store.
+                        DBCol::DbVersion,
+                        DBCol::BlockMisc,
+                        // Flat storage columns.
+                        DBCol::FlatState,
+                        DBCol::FlatStateChanges,
+                        DBCol::FlatStateDeltaMetadata,
+                        DBCol::FlatStorageStatus,
+                    ]),
+                )?;
+                let store = storage.get_hot_store();
+                let flat_storage_manager = FlatStorageManager::new(store.clone());
+
+                /*
+                let epoch_id = epoch_manager.get_epoch_id(prev_block_hash)?;
+                let shard_layout = epoch_manager.get_shard_layout(&epoch_id)?;
+                // TODO: Should FS be created only for the tracked shards?
+                let shard_uids = shard_layout.get_shard_uids();
+                 */
+
+                *state_snapshot_lock = Some(StateSnapshot::new(
+                    store,
+                    *prev_block_hash,
+                    flat_storage_manager,
+                    shard_uids,
+                ));
+                tracing::info!(target: "state_snapshot", ?prev_block_hash, "Made a checkpoint");
+                Ok(())
+            }
+        }
+    }
+
+    pub fn compact_state_snapshot(
+        &self,
+        prev_block_hash: &CryptoHash,
```

Comment:
Do we actually need to pass `prev_block_hash` here? Is it enough to request compaction without any args?